tests/testthat/test-thumbnail.R

```diff
@@ -0,0 +1,120 @@
+        
+        # User-specified path
+        user_path <- tempfile("thumbnail_", fileext = ".jpg")
+        received_path <- get_thumbnail(item, user_path)
+        expect_equal(received_path, NA_character_)
+        
+        # Automatic path
+        received_path <- get_thumbnail(item)
+        expect_equal(received_path, NA_character_)
+      })
+    
+      test_that(glue::glue("get_thumbnail() errors with 404 status codes ({version})"), {
+        client <- connect(server = "https://connect.example", api_key = "fake")
+        item <- content_item(client, "23456789")
+        expect_error(get_thumbnail(item), "request failed with Client error: \\(404\\) Not Found")
+      })
+    
+      test_that(glue::glue("has_thumbnail() returns TRUE when the item has a thumbnail ({version})"), {
+        client <- connect(server = "https://connect.example", api_key = "fake")
+        item <- content_item(client, "01234567")
+        expect_true(has_thumbnail(item))
+      })
+    
+      test_that(glue::glue("has_thumbnail() returns FALSE when the status code is 204 ({version})"), {
+        client <- connect(server = "https://connect.example", api_key = "fake")
+        item <- content_item(client, "12345678")
+        expect_false(has_thumbnail(item))
+      })
+    
+      test_that(glue::glue("has_thumbnail() errors with 404 status codes ({version})"), {
+        client <- connect(server = "https://connect.example", api_key = "fake")
+        item <- content_item(client, "23456789")
+        expect_error(has_thumbnail(item), "request failed with Client error: \\(404\\) Not Found")
+      })
+    
+      test_that(glue::glue("set_thumbnail() returns returns a content item when successful ({version})"), {
+        client <- connect(server = "https://connect.example", api_key = "fake")
+        item <- content_item(client, "01234567")
+        received <- set_thumbnail(item, "resources/smol.jpg")
+        expect_true(validate_R6_class(received, "Content"))
+        expect_identical(item, received)
+      })
+    
+      test_that(glue::glue("set_thumbnail() raises an error when the endpoint returns a 404 ({version})"), {
+        client <- connect(server = "https://connect.example", api_key = "fake")
+        item <- content_item(client, "23456789")
+        expect_error(set_thumbnail(item, "resources/smol.jpg"), "request failed with Client error: \\(404\\) Not Found")
+      })
+    
+      test_that(glue::glue("set_thumbnail() works with remote images ({version})"), {
+        client <- connect(server = "https://connect.example", api_key = "fake")
+        item <- content_item(client, "01234567")
+        expect_GET(set_thumbnail(item, "https://other.server/non-connect/working-image/not-an-image"),
+                   "https://other.server/non-connect/working-image/not-an-image")
+        # We're only asserting that this GETs.
+        # We can't use a mock because of this issue:
+        # https://github.com/nealrichardson/httptest/issues/86
```

Comment:
```suggestion
        # We're only asserting that the remote image hits GET. Because we httr::write_disk() and then using what is written doesn't work (https://github.com/nealrichardson/httptest/issues/86)
        # the rest of this function is tested elsewhere, so we have confidence that so long as getting and writing to disk works, we are good.
        expect_GET(set_thumbnail(item, "https://other.server/non-connect/working-image/not-an-image"),
                   "https://other.server/non-connect/working-image/not-an-image")
```

Reply:
Accepting suggestion locally so I can reflow the comment.